tools/webidl_binder.py

```diff
@@ -406,8 +406,19 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
   if not js_impl: continue
   implements[name] = [js_impl[0]]
 
+nodeHeight = {}
+for child, parent in implements.iteritems():
```

Comment:
please add a comment with this stuff, mainly "If the height of child is later increased, all its ancestors will be readjusted at that time to maintain that invariant. Further, the height of a node never decreases."